contracts/stellar-interchain-token-service/src/deployer.rs

```diff
@@ -84,7 +84,7 @@ pub fn deploy_token_manager(
         token_id,
         token_address,
         token_manager: deployed_address.clone(),
-        token_manager_type,
+        token_manager_type: token_manager_type.clone(),
```

Comment:
weird change. since you need a copy anyways, require that in the function argument so the behaviour is more clear to the caller. It's also consistent with what we do for other args.